src/stack/Stack.java

```diff
@@ -28,29 +32,47 @@ public void push(int value) {
 	 */
 	public int pop() {
 
-		return 0;
+		if (this.tos < 0) {
+			System.out.println("Stack is empty.");
+			return 0;
+		} else {
+			return data[tos--];
+		}
 	}
 
 
 	/**
 	 * Clear stack
 	 */
 	public void clear() {
-
+		if(!isEmpty()){
+			for(int i = 0; i < data.length; i++){
+				pop();
+			}
+			tos = -1;
+		}
 	}
 
 	/**
 	 * Check if stack is empty
 	 * @return
 	 */
 	public boolean isEmpty() {
-		return false;
+		if (this.tos < 0) {
+			return false;
+		} else {
+			return false;
+		}
 	}
 
 	/**
 	 * Increase the stack capacity if there is not enough space to add additional items
 	 */
-	private void ensureCapacity() {
+	private void ensureCapacity(int size) {
```

Comment:
The ensureCapacity() method can be written like this without  argument
private void ensureCapacity() {
		int[] newArray = new int[data.length * 2];
		System.arraycopy(data, 0, newArray, 0, data.length);
		this.data = newArray;
	}